roles/bwc/tasks/ldap.yml

```diff
@@ -0,0 +1,32 @@
+---
+
+- name: Setup st2.conf auth backend to LDAP
+  become: yes
+  # Unfortunately, ``with_dict`` also logs the dict which could leak passwords.
+  no_log: yes
+  ini_file:
+    dest: /etc/st2/st2.conf
+    section: auth
+    option: backend
+    value: ldap
+    backup: yes
+  # Don't even setup LDAP if backend_kwargs is not defined
+  with_dict: "{{ bwc_ldap.backend_kwargs | default({}) }}"
```

Comment:
I see `with_dict` here but don't see `{{ item }}`. That thing is used only for loops, see http://docs.ansible.com/ansible/playbooks_loops.html#looping-over-hashes.

So you probably want something like: 
```
when: bwc_ldap.backend_kwargs is defined
```